contracts/citycoin.clar

```diff
@@ -132,16 +135,47 @@
 )
 
 ;; Bind Stacks block height to a list of up to 32 miners (and how much they mined) per block,
-;; and track whether or not the miner has come back to claim their tokens.
-(define-map miners
+;; and track whether or not the miner has come back to claim their tokens and who mined the least.
+(define-map mined-blocks
     { stacks-block-height: uint }
     {
-        miners: (list 32 { miner: principal, amount-ustx: uint }),
-        claimed: bool
+        miners: (list 128 { miner-id: uint, amount-ustx: uint }),
+        claimed: bool,
+        least-commitment: (optional { miner-id: uint, amount-ustx: uint }),
     }
 )
 
-;; TO DO: think about adding amount to miners map.
+;; variable used to generate unique miner-id's
+(define-data-var miners-nonce uint u0)
+
+;; Maps miner address to uint miner-id
+(define-map miners
+    { miner: principal }
+    { miner-id: uint }
+)
+
+;; Returns miner ID if it has been created
+(define-read-only (get-miner-id (miner principal))
+    (get miner-id (map-get? miners { miner: miner }))
```

Comment:
This function returns `(optional uint)` type. If miner doesn't have id it will return `none`, otherwise it will return `(some uint)` for example `(some u123)`.